controllers/chat/changeChannelDetail.js

```diff
@@ -0,0 +1,44 @@
+const {StreamChat} = require('stream-chat');
+const {CHANNEL_TYPE_STRING} = require('../../helpers/constants');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const changeChannelDetail = async (req, res) => {
+  const {channel_id, channel_image, channel_name} = req.body;
+  const {userId} = req;
+
+  const client = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  const queriedChannel = await client.queryChannels({
+    type: CHANNEL_TYPE_STRING.GROUP,
+    id: channel_id,
+    members: {$in: [userId]}
+  });
+
+  if (queriedChannel?.length < 1) return ErrorResponse.e404(res, 'Channel not found');
+
+  const channel = await client.channel(CHANNEL_TYPE_STRING.GROUP, channel_id);
+  const updateData = {};
+  if (channel_image) updateData.image = channel_image;
+  if (channel_name) updateData.name = channel_name;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is checking for the existence of `channel_image` and `channel_name` before adding them to `updateData`. However, this could lead to a situation where neither `channel_image` nor `channel_name` are provided, and `updateData` remains an empty object. In such a case, the `updatePartial` function would be called with no actual data to update, which is unnecessary. You should add a check to ensure that at least one of `channel_image` or `channel_name` is provided.

```diff
+  if (!channel_image && !channel_name) {
+    return ErrorResponse.e400(res, 'At least one of channel_image or channel_name must be provided');
+  }
-  if (channel_image) updateData.image = channel_image;
-  if (channel_name) updateData.name = channel_name;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->